cli/internal/cmd/create.go

```diff
@@ -301,7 +303,7 @@ func must(err error) {
 }
 
 // validateCLIandConstellationVersionAreEqual checks if the image and microservice version are equal (down to patch level) to the CLI version.
-func validateCLIandConstellationVersionAreEqual(cliVersion, imageVersion, microserviceVersion string) error {
+func validateCLIandConstellationVersionAreEqual(cliVersion consemver.Semver, imageVersion string, microserviceVersion consemver.Semver) error {
```

Comment:
nit: should we make conf.Image a semver type too?

Reply:
I didn't because it behaves differently in a few situations. Namely, whenever it is a shortpath. I think the abstractions don't match very well and we should have a `ImageVersion`, `ShortPath` or similar type.